bitcoin/src/crypto/sighash.rs

```diff
@@ -869,6 +871,25 @@ impl<R: Borrow<Transaction>> SighashCache<R> {
         Ok(SegwitV0Sighash::from_engine(enc))
     }
 
+    /// Computes the BIP143 sighash for any flag type.
+    pub fn segwit_script_code_signature_hash(
+        &mut self,
+        input_index: usize,
+        script_code: &SegwitScriptCode,
+        value: Amount,
+        sighash_type: EcdsaSighashType,
+    ) -> Result<SegwitV0Sighash, Error> {
```

Comment:
In a76a190093aaad7c9a3b3d1160956f90974649ee:

This method should be infallible. You can put an `.expect` on the call to `segwit_encode_signing_data_to` since we know that encoding stuff to a hash engine cannot fail.